### ISSUE TITLE
Improve how image group name is computed in the tweakreg step

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#8011]
_Issue [JP-3442](https://jira.stsci.edu/browse/JP-3442) was created on JIRA by [Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara):_

Current algorithm is quite simple and does not look for duplicate names and has a check to ensure that input file names have equal length. However, some users may want to  use different length file names. See help desk ticket INC0194239 for more details.

CC: [Paul Goudfrooij](https://jira.stsci.edu/secure/ViewProfile.jspa?name=goudfroo)

**Comment by stscijgbot-jp** [comment]
_Comment by [Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara) on [JIRA](https://jira.stsci.edu/browse/JP-3442?focusedCommentId=667987#comment-667987):_

#8012

**Comment by stscijgbot-jp** [comment]
_Comment by [Rosa Diaz](https://jira.stsci.edu/secure/ViewProfile.jspa?name=rdiaz) on [JIRA](https://jira.stsci.edu/browse/JP-3442?focusedCommentId=673785#comment-673785):_

Here input from Paul:

Hello. I'm trying to re-run calwebb_image3 on a set of NIRCam short-wave "_cal" images, some of which I first ran through a script of mine to correct for some issue with the structure of the sky background (that's a separate issue). The output images from that script got a different prefix to indicate that they were run through that script of mine (say from _cal.fits to _cal_corr.fits). Then I try to run calwebb_image3 on the full set of dithered images (so some files are _cal.fits and some others are _cal_corr.fits), but it fails in the tweakreg step when it tries to list the images in a (dither) group. The traceback is as follows: 


Traceback (most recent call last): 
File "/Users/goudfroo/lib/python/pgimage3_custom_nrc.py", line 130, in <module> 
result3 = level3.call( 
File "/Users/goudfroo/miniconda3/envs/jwstpipe/lib/python3.10/site-packages/stpipe/step.py", line 652, in call 
return instance.run(*args) 
File "/Users/goudfroo/miniconda3/envs/jwstpipe/lib/python3.10/site-packages/stpipe/step.py", line 478, in run 
step_result = self.process(*args) 
File "/Users/goudfroo/miniconda3/envs/jwstpipe/lib/python3.10/site-packages/jwst/pipeline/calwebb_image3.py", line 89, in process 
input_models = self.tweakreg(input_models) 
File "/Users/goudfroo/miniconda3/envs/jwstpipe/lib/python3.10/site-packages/stpipe/step.py", line 478, in run 
step_result = self.process(*args) 
File "/Users/goudfroo/miniconda3/envs/jwstpipe/lib/python3.10/site-packages/jwst/tweakreg/tweakreg_step.py", line 283, in process 
group_name = _common_name(g) 
File "/Users/goudfroo/miniconda3/envs/jwstpipe/lib/python3.10/site-packages/jwst/tweakreg/tweakreg_step.py", line 548, in _common_name 
assert all(fname_len[0] == length for length in fname_len) 
AssertionError 

So I looked in that script "tweakreg_step.py", which I see was written by Mihai Cara, and I see this function _common_name: 

def _common_name(group): 
file_names = [path.splitext(im.meta.filename)[0].strip('_- ') 
for im in group] 
fname_len = list(map(len, file_names)) 
assert all(fname_len[0] == length for length in fname_len) 
cn = path.commonprefix(file_names) 
assert cn 
return cn 

so.... do I understand correctly that all file names in a given (dither) group must have the same length????? I wonder why that is required? I guess that would mean that one has to make sure all input file names to calwebb_image3 have the same length when running calwebb_image3, which seems a bit restrictive?

**Comment by stscijgbot-jp** [comment]
_Comment by [Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara) on [JIRA](https://jira.stsci.edu/browse/JP-3442?focusedCommentId=673791#comment-673791):_

This input from Paul is a copy and paste of the text from the ticket [INC0194239](https://stsci.service-now.com/nav_to.do?uri=incident.do%3Fsys_id=e4ebec0c97067d1044d5f6a3f153af27%26sysparm_stack=incident_list.do%3Fsysparm_query=active=true) opened on October 18th. My PR listed above from October 18th should solve this issue. Are you saying that my PR still did not solve the issue?

**Comment by stscijgbot-jp** [comment]
_Comment by [Rosa Diaz](https://jira.stsci.edu/secure/ViewProfile.jspa?name=rdiaz) on [JIRA](https://jira.stsci.edu/browse/JP-3442?focusedCommentId=673821#comment-673821):_

[Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara)  I should edit my comment to say that it is for documentation purposes only, so it is easier to know what the instrument teams are supposed to test and also later for the release notes. I assume your fix works but want to let the INS testers to check if they agree.

**Comment by stscijgbot-jp** [comment]
_Comment by [Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara) on [JIRA](https://jira.stsci.edu/browse/JP-3442?focusedCommentId=673849#comment-673849):_

Oh, I see. Sorry

**Comment by stscijgbot-jp** [comment]
_Comment by [Misty Cracraft](https://jira.stsci.edu/secure/ViewProfile.jspa?name=cracraft) on [JIRA](https://jira.stsci.edu/browse/JP-3442?focusedCommentId=674309#comment-674309):_

For MIRI Imaging, to test this update, I created an association file with a set of *_cal.fits where I had added in numbers, 2, 23, and 234 to the filenames just to change the name length. The association file ran through calwebb_image3 with no problems, and for the F770W test dataset, tweakreg had no problems finding sources and aligning the images. From my understanding, this tested the ticket and shows that the fix works fine.
